lib/APPReinforcementLearning/src/App.h

```diff
@@ -97,22 +115,49 @@ class App
     SimpleTimer m_controlInterval;
 
     /**
-     * Copy construction of an instance.
-     * Not allowed.
-     *
-     * @param[in] app Source instance.
-     */
-    App(const App& app);
+    * Timer for sending system status to DCS.
+    */
+    SimpleTimer m_statusTimer;
+
+    /** Timer used for sending data periodically. */
+    SimpleTimer m_sendLineSensorsDataInterval;
 
+    /** Send status timer interval in ms. */
+    static const uint32_t SEND_STATUS_TIMER_INTERVAL = 1000U;
+
+    /** Sending Data period in ms. */
+    static const uint32_t SEND_LINE_SENSORS_DATA_PERIOD = 20U;
+
+    /** SerialMuxProt Channel id for sending system status. */
+    uint8_t m_serialMuxProtChannelIdStatus;
+
+    /** SerialMuxProt Channel id for sending LineSensors. */
+    uint8_t m_serialMuxProtChannelIdLineSensors;
+
+    /** SerialMuxProt Channel id for sending Mode Selected. */
+    uint8_t m_serialMuxProtChannelIdMode;
+
+    /** SerialMuxProt Server Instance. */
+    SMPServer m_smpServer;
+
+    /* Ensue that the mode is only sent once*/
+    bool m_sentmode;
+   
     /**
-     * Assignment of an instance.
-     * Not allowed.
-     *
-     * @param[in] app Source instance.
+     * Setup the SerialMuxProt channels.
      *
-     * @returns Reference to App instance.
+     * @return If successful returns true, otherwise false.
      */
-    App& operator=(const App& app);
+    bool setupSerialMuxProt();
+    
+    /**
+     * Send line sensors data via SerialMuxProt.
+    */
+    void sendLineSensorsData() const;
+
+    /* Not allowed. */
+    App(const App& app);            /**< Copy construction of an instance. */
+    App& operator=(const App& app); /**< Assignment of an instance. */
```

Comment:
Why modified the comment of these methods? This is old, please change back to the new comment